test/hierarchical_bitset/main.cpp

```diff
@@ -0,0 +1,169 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2021 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#include <osp/hierarchical_bitset.h>
+
+#include <gtest/gtest.h>
+
+#include <array>
+#include <random>
+#include <vector>
+
+using osp::HierarchicalBitset;
+
+/**
+ * @brief Generate random integers in ascending order up to a certain maximum
+ */
+std::vector<int> random_ascending(int seed, size_t maximum)
+{
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<int> dist(0, 1);
+    std::vector<int> testSet;
+
+    for (size_t i = 0; i < maximum; i ++)
+    {
+        if (dist(gen) == 1)
+        {
+            testSet.push_back(i);
+        }
+    }
+
+    return testSet;
+}
+
+TEST(HierarchicalBitset, CountTrailingZeros)
+{
+    using osp::ctz;
+
+    EXPECT_EQ( 0,  ctz(uint64_t(1)) );
+    EXPECT_EQ( 20, ctz(uint64_t(1) << 20) );
+    EXPECT_EQ( 63, ctz(uint64_t(1) << 63) );
+}
+
+TEST(HierarchicalBitset, BasicUnaligned)
```

Comment:
If you could, some documentation before each test of what's being tested and why would be helpful.